File: 268-missing-number/missing-number.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n = nums.size();
        int Expected_Sum = n*(n+1)/2;
        int Current_Sum = 0;
        for(int i=0;i<n;i++){
            Current_Sum += nums[i];
        }
        int ans = Expected_Sum - Current_Sum;
        return ans;
    }
};

File: 268-missing-number/missing-number.java
class Solution {
    public int missingNumber(int[] nums) {
        int ans = -1;
        HashMap<Integer,Integer>mpp = new HashMap<>();
        for(int i : nums){
            mpp.put(i,mpp.getOrDefault(i,0)+1);
        }
        for(int i = 0;i<=nums.length;i++){
            if(!mpp.containsKey(i)){
                return i;
            }
        }
        return ans;
    }
}